apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    private static void mockTargetWorkspace(String apiKey, String workspaceName, String workspaceId) {
+        AuthTestUtils.mockTargetWorkspace(wireMock.server(), apiKey, workspaceName, workspaceId, workspaceId);
+    }
+
+    @Nested
+    @DisplayName("Get Feedback Score names")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class GetFeedbackScoreNames {
+
+        @ParameterizedTest
+        @MethodSource
+        @DisplayName("when get feedback score names, then return feedback score names")
+        void getFeedbackScoreNames__whenGetFeedbackScoreNames__thenReturnFeedbackScoreNames(
+                boolean userProjectId,
+                boolean withExperimentsOnly,
+                List<String> names,
+                List<String> otherNames) {
+            // given
+            var apiKey = UUID.randomUUID().toString();
+            var workspaceId = UUID.randomUUID().toString();
+            var workspaceName = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            // when
+            String projectName = UUID.randomUUID().toString();
+
+            UUID projectId = projectResourceClient.createProject(projectName, apiKey, workspaceName);
+            Project project = projectResourceClient.getProject(projectId, apiKey, workspaceName);
+
+            // Create multiple values feedback scores
+            List<String> multipleValuesFeedbackScores = names.subList(0, names.size() - 1);
+
+            List<List<FeedbackScoreBatchItem>> multipleValuesFeedbackScoreList = createMultiValueScores(
+                    multipleValuesFeedbackScores, project, apiKey, workspaceName);
+
+            List<List<FeedbackScoreBatchItem>> singleValueScores = createMultiValueScores(List.of(names.getLast()),
+                    project, apiKey, workspaceName);
+
+            createExperimentsItems(apiKey, workspaceName, multipleValuesFeedbackScoreList, singleValueScores);
+
+            // Create unexpected feedback scores
+            var unexpectedProject = podamFactory.manufacturePojo(Project.class);
+
+            List<List<FeedbackScoreBatchItem>> unexpectedScores = createMultiValueScores(otherNames, unexpectedProject,
+                    apiKey, workspaceName);
+
+            if (!withExperimentsOnly) {
+                createExperimentsItems(apiKey, workspaceName, unexpectedScores, List.of());
+            }
+
+            fetchAndAssertResponse(userProjectId, withExperimentsOnly, names, otherNames, projectId, apiKey,
+                    workspaceName);
+        }
+
+        Stream<Arguments> getFeedbackScoreNames__whenGetFeedbackScoreNames__thenReturnFeedbackScoreNames() {
+            return Stream.of(
+                    Arguments.of(true, false,
+                            PodamFactoryUtils.manufacturePojoList(podamFactory, String.class),
+                            PodamFactoryUtils.manufacturePojoList(podamFactory, String.class)),
```

Comment:
I don't see a reason why these should be passed as arguments. Can we move them to the test body?

Reply:
Good point!